### PR TITLE
At7/ld87

### PR DESCRIPTION
**Pull request by at7**
Make population name a required parameter for LD/id and LD/region endpoints.

### CODE REVIEW

File: lib/EnsEMBL/REST/Model/LDFeatureContainer.pm [hunk 1]
```diff
@@ -64,27 +63,23 @@ sub fetch_LDFeatureContainer_variation_name {
...
-    }
-    my $ldfc;
-    try {
-      $ldfc = $ldfca->fetch_by_VariationFeature($vf, $population);
-    } catch {
-      Catalyst::Exception->throw("Something went wrong while fetching from LDFeatureContainerAdaptor");
-    };
-    return $self->to_array($ldfc)
+  my $pa = $c->model('Registry')->get_adaptor($species, 'Variation', 'Population');     
+  my $population = $pa->fetch_by_name($population_name);
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: Can this throw Ensembl errors?

File: lib/EnsEMBL/REST/Model/LDFeatureContainer.pm [hunk 2]
```diff
@@ -99,23 +94,18 @@ sub fetch_LDFeatureContainer_slice {
...
+  if (!$population) {
+    Catalyst::Exception->throw("Could not fetch population object for population name: $population_name");
   }
-  my $ldfc = $ldfca->fetch_by_Slice($slice);
-  return $self->to_array($ldfc);
+  my $ldfc;
+  try {
+    $ldfc = $ldfca->fetch_by_Slice($slice, $population);
+  } catch {
+    Catalyst::Exception->throw("Something went wrong while fetching from LDFeatureContainerAdaptor");
```
Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: Does the ldfca error make it to the log files, or have you swallowed it here on purpose?

File: lib/EnsEMBL/REST/Model/LDFeatureContainer.pm (hunk 1 above)
Comment (at7) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: The PopulationAdaptor returns undef if the population isn't stored in our database.
population_name is now required for this endpoint. I already check in the controller if population_name is given and throw an exception if not.

File: lib/EnsEMBL/REST/Model/LDFeatureContainer.pm (hunk 2 above)
Comment (at7) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: How do I write to the log files?

Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: I presume you can find the logger through the context - $c->log->error($_)

Comment (nerdstrike) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: Otherwise we just "something went wrong" and we can't debug it.

Comment (at7) [inline lib/EnsEMBL/REST/Model/LDFeatureContainer.pm]: I'll add that. thanks

**Comment by nerdstrike** [comment]
I think this is ok for integration now

**Pull request merged**